docs/authentication/stytch/permit-integration.mdx

```diff
@@ -0,0 +1,349 @@
+  so [here](https://stytch.com/docs/guides/quickstarts/nextjs#before-you-start).
+- Permit account
+- Basic understanding of Next.js App Router
+
+## Stytch Step-by-Step Guide
+
+### Step 1: Install Stytch SDKs and Configure Your API Keys
+
+First, add our frontend JavaScript SDK and backend Node SDK packages to your Next.js application:
+
+```bash
+npm install @stytch/nextjs @stytch/vanilla-js stytch --save
+```
+
+Next, add your Stytch project's API keys to your application as environment variables:
+
+```
+STYTCH_PROJECT_ENV=test
+STYTCH_PROJECT_ID="YOUR_STYTCH_PROJECT_ID"
+STYTCH_PUBLIC_TOKEN="YOUR_STYTCH_PUBLIC_TOKEN"
+STYTCH_SECRET="YOUR_STYTCH_SECRET"
+```
+
+### Step 2: Wrap Your Application in a StytchProvider
+
+Initialize the Stytch client and add a StytchProvider component to the root of your application so that the frontend JavaScript
+SDK is available to child components:
+
+```javascript
+import { StytchProvider, createStytchUIClient } from "@stytch/nextjs";
+import Head from "next/head";
+
+const stytch = createStytchUIClient(process.env.STYTCH_PUBLIC_TOKEN);
+
+export default function App({ Component, pageProps }) {
+  return (
+    <>
+      <Head>{/* Truncated */}</Head>
+      <StytchProvider stytch={stytch}>
+        <Component {...pageProps} />
+      </StytchProvider>
+    </>
+  );
+}
+```
+
+### Step 3: Add a StytchLogin Component
+
+Create a StytchLogin component. You can configure which authentication methods you'd like to offer by modifying the products array
+in your config object. Here's an example that includes Email Magic Links and OAuth products:
+
+```javascript
+import { StytchLogin, Products } from "@stytch/vanilla-js";
+
+const Login = () => {
+  const REDIRECT_URL = "SEE_STEP_5";
+
+  const config = {
+    products: [Products.emailMagicLinks, Products.oauth],
+    emailMagicLinksOptions: {
+      loginRedirectURL: REDIRECT_URL,
+      loginExpirationMinutes: 60,
+      signupRedirectURL: REDIRECT_URL,
+      signupExpirationMinutes: 60,
+    },
+    oauthOptions: {
+      providers: [{ type: "google" }],
+      loginRedirectURL: REDIRECT_URL,
+      signupRedirectURL: REDIRECT_URL,
+    },
+  };
+
+  return <StytchLogin config={config} styles={{}} />;
+};
+```
+
+Next, add the StytchLogin component to your application's login page. You'll notice that we check for a logged-in user
+before displaying the StytchLogin component:
+
+```javascript
+import { useStytchUser } from "@stytch/nextjs";
+import { useRouter } from "next/router";
+import { useEffect } from "react";
+
+export default function LoginPage() {
+  const { user, isInitialized } = useStytchUser();
+  const router = useRouter();
+
+  useEffect(() => {
+    if (isInitialized && user) {
+      router.replace("/profile");
+    }
+  }, [user, isInitialized, router]);
+
+  return <Login />;
+}
+```
+
+### Step 4: Handle Redirects and Complete the Authentication Flow
+
```

Comment:
This step is unclear to me, I think it should be re-written, maybe split the code or write comments in it.